### PR TITLE
refactor: move default RGB/LED matrix #defines

### PR DESCRIPTION
**Pull request by lesshonor**
## Description

Moving the fallback definitions of macros like `LED_MATRIX_VAL_STEP` and `RGB_MATRIX_MAXIMUM_BRIGHTNESS` to header files allows keyboards to leverage these defaults without requiring `#ifdef` guards (and often repeating said fallback definitions).

...also corrected a typo in the console output for `led_matrix_set_flags_eeprom_helper()`/`rgb_matrix_set_flags_eeprom_helper()`.

---

Generated SHAs for `default` and `via` keymaps using `qmk mass-compile -f 'exists(led_matrix.driver)' -e SKIP_VERSION=yes` and `qmk mass-compile -f 'exists(rgb_matrix.driver)' -e SKIP_VERSION=yes` (...plus `teleport/native`'s `perfmode` keymaps) to verify this would not change existing behavior.

Out of 1396, there were 9 mismatches:

1. `junco/rev1:default`, `keebio/convolution/rev1:default`, `keychron/v3/iso:default`,`keychron/v3/iso:via`, `moonlander:default`, `moonlander:via` all have console enabled, so were impacted by the typo fix.
2. `kbdfans/kbd67/mkiirgb/v3:default`, `kbdfans/kbd67/mkiirgb/v3:via` and `zykrah/fuyu:via` all manipulate a caps lock indicator based on `RGB_MATRIX_MAXIMUM_BRIGHTNESS`/`RGB_MATRIX_VAL_STEP`. 
   The way the existing code was written, the indicators ended up with a value step of 8 while the rest of the lights defaulted to 16. It seems clear that the intent was to scale the indicator *with* everything else, so the affected keymaps/keyboard were modified accordingly. @zykrah @moyi4681 

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: quantum/led_matrix/led_matrix.h [hunk 1]
```diff
@@ -37,6 +37,35 @@
...
+#    define LED_MATRIX_DEFAULT_MODE LED_MATRIX_SOLID
+#endif
+
+#if !defined(LED_MATRIX_DEFAULT_VAL)
+#    define LED_MATRIX_DEFAULT_VAL LED_MATRIX_MAXIMUM_BRIGHTNESS
+#endif
+
+#if !defined(LED_MATRIX_DEFAULT_SPD)
+#    define LED_MATRIX_DEFAULT_SPD UINT8_MAX / 2
+#endif
```
Comment (zvecr) [inline quantum/led_matrix/led_matrix.h:67]: `LED_MATRIX_MAXIMUM_BRIGHTNESS > UINT8_MAX` check no longer adds value, as you get a build error "Invalid API data: max_brightness: 400 is greater than the maximum of 255".

```suggestion
#ifndef LED_MATRIX_MAXIMUM_BRIGHTNESS
#    define LED_MATRIX_MAXIMUM_BRIGHTNESS UINT8_MAX
#endif

#ifndef LED_MATRIX_VAL_STEP
#    define LED_MATRIX_VAL_STEP 8
#endif

#ifndef LED_MATRIX_SPD_STEP
#    define LED_MATRIX_SPD_STEP 16
#endif

#ifndef LED_MATRIX_DEFAULT_MODE
#    define LED_MATRIX_DEFAULT_MODE LED_MATRIX_SOLID
#endif

#ifndef LED_MATRIX_DEFAULT_VAL
#    define LED_MATRIX_DEFAULT_VAL LED_MATRIX_MAXIMUM_BRIGHTNESS
#endif

#ifndef LED_MATRIX_DEFAULT_SPD
#    define LED_MATRIX_DEFAULT_SPD UINT8_MAX / 2
#endif
```

File: quantum/rgb_matrix/rgb_matrix.h [hunk 2]
```diff
@@ -44,6 +44,56 @@
...
+#    define RGB_MATRIX_DEFAULT_SAT UINT8_MAX
+#endif
+
+#if !defined(RGB_MATRIX_DEFAULT_VAL)
+#    define RGB_MATRIX_DEFAULT_VAL RGB_MATRIX_MAXIMUM_BRIGHTNESS
+#endif
+
+#if !defined(RGB_MATRIX_DEFAULT_SPD)
+#    define RGB_MATRIX_DEFAULT_SPD UINT8_MAX / 2
+#endif
```
Comment (zvecr) [inline quantum/rgb_matrix/rgb_matrix.h:95]:
```suggestion
#ifndef RGB_MATRIX_MAXIMUM_BRIGHTNESS
#    define RGB_MATRIX_MAXIMUM_BRIGHTNESS UINT8_MAX
#endif

#ifndef RGB_MATRIX_HUE_STEP
#    define RGB_MATRIX_HUE_STEP 8
#endif

#ifndef RGB_MATRIX_SAT_STEP
#    define RGB_MATRIX_SAT_STEP 16
#endif

#ifndef RGB_MATRIX_VAL_STEP
#    define RGB_MATRIX_VAL_STEP 16
#endif

#ifndef RGB_MATRIX_SPD_STEP
#    define RGB_MATRIX_SPD_STEP 16
#endif

#ifndef RGB_MATRIX_DEFAULT_MODE
#    ifdef ENABLE_RGB_MATRIX_CYCLE_LEFT_RIGHT
#        define RGB_MATRIX_DEFAULT_MODE RGB_MATRIX_CYCLE_LEFT_RIGHT
#    else
// fallback to solid colors if RGB_MATRIX_CYCLE_LEFT_RIGHT is disabled in userspace
#        define RGB_MATRIX_DEFAULT_MODE RGB_MATRIX_SOLID_COLOR
#    endif
#endif

#ifndef RGB_MATRIX_DEFAULT_HUE
#    define RGB_MATRIX_DEFAULT_HUE 0
#endif

#ifndef RGB_MATRIX_DEFAULT_SAT
#    define RGB_MATRIX_DEFAULT_SAT UINT8_MAX
#endif

#ifndef RGB_MATRIX_DEFAULT_VAL
#    define RGB_MATRIX_DEFAULT_VAL RGB_MATRIX_MAXIMUM_BRIGHTNESS
#endif

#ifndef RGB_MATRIX_DEFAULT_SPD
#    define RGB_MATRIX_DEFAULT_SPD UINT8_MAX / 2
#endif
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/teleport/native/config.h [hunk 3]
```diff
@@ -61,12 +61,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #define ENABLE_RGB_MATRIX_CUSTOM_SINGLE_COLOR_RAINDROPS // 0 Single color raindrops, random keys lighting up at randomized intensity
 #define ENABLE_RGB_MATRIX_CUSTOM_STATIC_GAME_MODE       // - Game mode sets the entire matrix (static) once, then stops LED refreshes
 
-#define INDICATOR_MAX_BRIGHTNESS 255
-#ifdef RGB_MATRIX_MAXIMUM_BRIGHTNESS
-    #undef INDICATOR_MAX_BRIGHTNESS
-    #define INDICATOR_MAX_BRIGHTNESS RGB_MATRIX_MAXIMUM_BRIGHTNESS
-#endif
-
+#define INDICATOR_MAX_BRIGHTNESS RGB_MATRIX_MAXIMUM_BRIGHTNESS
```
Comment (zvecr) [inline keyboards/teleport/native/config.h:64]: I guess this doesnt add a ton of value,
```suggestion
```

And then just update the following places to use `RGB_MATRIX_MAXIMUM_BRIGHTNESS`?
```
keyboards/teleport/native/ansi/keymaps/via/keymap.c
60:    HSV hsv_ind = {rgb_matrix_get_hue()+30,255,INDICATOR_MAX_BRIGHTNESS};
73:      HSV hsv_ind = {rgb_matrix_get_hue()+30,255,INDICATOR_MAX_BRIGHTNESS};

keyboards/teleport/native/iso/keymaps/via/keymap.c
60:    HSV hsv_ind = {rgb_matrix_get_hue()+30,255,INDICATOR_MAX_BRIGHTNESS};
73:      HSV hsv_ind = {rgb_matrix_get_hue()+30,255,INDICATOR_MAX_BRIGHTNESS};

keyboards/teleport/native/iso/keymaps/perfmode/keymap.c
153:    HSV hsv_ind = {rgb_matrix_get_hue()+30,255,INDICATOR_MAX_BRIGHTNESS};
166:      HSV hsv_ind = {rgb_matrix_get_hue()+30,255,INDICATOR_MAX_BRIGHTNESS};

keyboards/teleport/native/ansi/keymaps/default/keymap.c
60:    HSV hsv_ind = {rgb_matrix_get_hue()+30,255,INDICATOR_MAX_BRIGHTNESS};
73:      HSV hsv_ind = {rgb_matrix_get_hue()+30,255,INDICATOR_MAX_BRIGHTNESS};

keyboards/teleport/native/iso/keymaps/default/keymap.c
60:    HSV hsv_ind = {rgb_matrix_get_hue()+30,255,INDICATOR_MAX_BRIGHTNESS};
73:      HSV hsv_ind = {rgb_matrix_get_hue()+30,255,INDICATOR_MAX_BRIGHTNESS};

keyboards/teleport/native/ansi/keymaps/perfmode/keymap.c
153:    HSV hsv_ind = {rgb_matrix_get_hue()+30,255,INDICATOR_MAX_BRIGHTNESS};
166:      HSV hsv_ind = {rgb_matrix_get_hue()+30,255,INDICATOR_MAX_BRIGHTNESS};
```

**Review by dexter93 — APPROVED** [review]
this has been a long time coming. Yes please

**Review by Xelus22 — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by zvecr — APPROVED** [review]

**Pull request merged**